### ISSUE TITLE
False positive undefined_name with doubly-nested class

### ISSUE THREAD

**Issue by JelleZijlstra** [quora/pyanalyze#733]
```python
def f():
    y = 3

    def inner():

        class Nested:
            classvar = y
```

Produces:

```
Undefined name: y (code: undefined_name)
In nestedcls.py at line 7
   4:     def inner():
   5: 
   6:         class Nested:
   7:             classvar = y
                             ^
```
